sendgrid.go

```diff
@@ -32,6 +41,7 @@ func createSendGridRequest(sgOptions sendGridOptions) rest.Request {
 		sgOptions.Endpoint,
 		sgOptions.Host,
 		sgOptions.Subuser,
+		sgOptions.Region,
```

Comment:
you can add a setter for host. In case, the users want to set host if they want to.